### PR TITLE
[router] 4.14 Enforce a recommended minimum number of required confirmations

### LINKED ISSUE #161: [router] 4.14 Use safe confirmations

### PR DESCRIPTION
**Pull request by jakekidd**
Fixes #161 
~~Blocked by https://github.com/connext/chaindata/pull/29~~

### CODE REVIEW

File: packages/router/src/config.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ///NXTP Config Generator based on vector/modules/router/src/config.ts
+import https from "https";
```
Comment (rhlsthrm) [inline packages/router/src/config.ts:2]: Ethers already has a `fetchJson` method that is better for this use case.

File: packages/router/src/config.ts [hunk 2]
```diff
@@ -24,6 +25,43 @@ import {
 } from "@connext/nxtp-utils";
 import { config as dotenvConfig } from "dotenv";
 
+// TODO: Should we be caching chain data locally? Like, request at most once per day, or once per week?
+// Maybe there could be a versioning system of this remote config, and we can use that to check whether our
+// local cache is up to date?
+const CHAIN_DATA = new Promise<Map<string, any>>((resolve) => {
+  // TODO: Using raw github api is not a stable solution.
```
Comment (rhlsthrm) [inline packages/router/src/config.ts:32]: Why not? I think this is a fine way to do it.

**Review by rhlsthrm — CHANGES_REQUESTED** [review]

File: packages/router/src/config.ts (hunk 1 above)
Comment (jakekidd) [reply packages/router/src/config.ts:2]: I almost used that method but I was concerned bc the method seemed to be doing unnecessary work for our case

File: packages/router/src/config.ts (hunk 2 above)
Comment (jakekidd) [reply packages/router/src/config.ts:32]: Okay - I was worried that relying on github to host adds the constraint that, if github stops working, our routers stop working. Github probs not gonna stop working lol, but if it does, would suck.

I'll update this to be a TODO with the above statement regarding caching the remote config, and checking for updates. If github is randomly down, we can just revert to cached.

Comment (jakekidd) [reply packages/router/src/config.ts:32]: Should probably actually go ahead and do that in this PR...

Comment (rhlsthrm) [reply packages/router/src/config.ts:32]: We should be able to default to something if it's not working.

Comment (rhlsthrm) [reply packages/router/src/config.ts]: Actually I'm not sure. This should only happen on startup anyways so I think it's okay to crash the router if this doesn't work.

File: packages/router/src/config.ts [hunk 3]
```diff
@@ -22,6 +22,39 @@ import {
...
+  const chainDataToMap = (data: any) => {
+    const chainData: Map<string, any> = new Map();
+    for (let i = 0; i < data.length; i++) {
+      const item = data[i];
+      const chainId = item.chainId.toString();
+      chainData.set(chainId, Object.fromEntries(Object.entries(item).filter((e) => e[0] !== "chainId")));
+    }
+    return chainData;
+  };
+  fetchJson(url)
```
Comment (rhlsthrm) [inline packages/router/src/config.ts:39]: Why no
```ts
try {
  CHAIN_DATA = await fetchJson(...)
} catch (err) {
  ...
}
```
?

**Review by rhlsthrm — CHANGES_REQUESTED** [review]
I think there's an easier way to do this. I'm actually doing this in my other PR, will share it with you when we chat.

**Comment by jakekidd** [comment]
Current error in integration

```
[app] [1629353763330] ERROR (0xDc150c5Db2cD1d1d8e505F824aBd90aEF887caC6/17073 on DESKTOP-9VH1R8A): Error in auction request
[app]     requestContext: {
[app]       "id": "jOv8W7wOQL+iJO/HaWnsJQ/0",
[app]       "origin": "subscribeToAuctionRequest"
[app]     }
[app]     err: {
[app]       "message": "No provider was found for chain 1337! Make sure this chain's providers are configured.",
[app]       "context": {},
[app]       "type": "NxtpError",
[app]       "stack":
[app]           Error: No provider was found for chain 1337! Make sure this chain's providers are configured.
[app]               at TransactionService.getProvider (/home/jakekidd/nxtp/packages/txservice/dist/txservice.js:273:19)
[app]                   -> /home/jakekidd/nxtp/packages/txservice/src/txservice.ts:327:13
[app]               at TransactionService.getDecimalsForAsset (/home/jakekidd/nxtp/packages/txservice/dist/txservice.js:188:35)
[app]                   -> /home/jakekidd/nxtp/packages/txservice/src/txservice.ts:224:31
[app]               at newAuction (/home/jakekidd/nxtp/packages/router/dist/lib/operations/auction.js:70:19)
[app]                   -> /home/jakekidd/nxtp/packages/router/src/lib/operations/auction.ts:106:15
[app]               at Object.auctionRequestBinding (/home/jakekidd/nxtp/packages/router/dist/bindings/messaging/auctionRequest.js:17:41)
[app]                   -> /home/jakekidd/nxtp/packages/router/src/bindings/messaging/auctionRequest.ts:21:39
[app]               at /home/jakekidd/nxtp/packages/router/dist/bindings/messaging/index.js:15:36
[app]                   -> /home/jakekidd/nxtp/packages/router/src/bindings/messaging/index.ts:17:13
[app]               at /home/jakekidd/nxtp/packages/utils/dist/messaging.js:381:20
[app]                   -> /home/jakekidd/nxtp/packages/utils/src/messaging.ts:487:16
[app]               at /home/jakekidd/nxtp/packages/utils/dist/messaging.js:363:20
[app]                   -> /home/jakekidd/nxtp/packages/utils/src/messaging.ts:465:14
[app]               at /home/jakekidd/nxtp/packages/utils/dist/messaging.js:200:13
[app]                   -> /home/jakekidd/nxtp/packages/utils/src/messaging.ts:231:7
[app]               at Object.callback (/home/jakekidd/nxtp/node_modules/ts-natsutil/dist/cjs/nats.js:133:115)
[app]                   -> /home/jakekidd/nxtp/node_modules/ts-natsutil/src/nats.ts:131:59
[app]               at ProtocolHandler.processMsg (/home/jakekidd/nxtp/node_modules/ts-nats/lib/protocolhandler.js:874:25)
[app]                   -> /home/jakekidd/nxtp/node_modules/src/protocolhandler.ts:951:25
[app]     }
```

**Pull request merged**